### PR TITLE
Doc: Added warning to treat group chunks as immutable when using apply

### PR DESCRIPTION
**Pull request by pdpark**
- [X] closes issue #14180

### CODE REVIEW

File: doc/source/gotchas.rst [hunk 1]
```diff
@@ -332,3 +332,97 @@ using something similar to the following:
 See `the NumPy documentation on byte order
 <https://docs.scipy.org/doc/numpy/user/basics.byteswapping.html>`__ for more
 details.
+
+
+Alternative to storing lists in Pandas DataFrame Cells
```
Comment (jreback) [inline doc/source/gotchas.rst]: just DataFrame

File: doc/source/gotchas.rst [hunk 2]
```diff
@@ -332,3 +332,97 @@ using something similar to the following:
...
+------------------------------------------------------
+Storing nested lists/arrays inside a pandas object should be avoided for performance and memory use reasons. Instead they should be "exploded" into a flat DataFrame structure.
```
Comment (jreback) [inline doc/source/gotchas.rst]: use double backticks around ``DataFrame``

File: doc/source/gotchas.rst [hunk 3]
```diff
@@ -332,3 +332,97 @@ using something similar to the following:
...
+
+Example of exploding nested lists into a DataFrame:
+
+.. ipython:: python
+
+   from collections import OrderedDict
+   df = (pd.DataFrame(OrderedDict([('name', ['A.J. Price']*3),
```
Comment (jreback) [inline doc/source/gotchas.rst]: use dict contruction directly, if you want column ordering then pass columns

File: doc/source/gotchas.rst [hunk 4]
```diff
@@ -332,3 +332,97 @@ using something similar to the following:
...
+
+   from collections import OrderedDict
+   df = (pd.DataFrame(OrderedDict([('name', ['A.J. Price']*3), 
+                                   ('opponent', ['76ers', 'blazers', 'bobcats']), 
+                                   ('attribute x', ['A','B','C'])
+                                  ])
+                     ))
+   df
+
+   nn = [['Zach LaVine', 'Jeremy Lin', 'Nate Robinson', 'Isaia']]*3
```
Comment (jreback) [inline doc/source/gotchas.rst]: call this something more apparent

File: doc/source/gotchas.rst [hunk 5]
```diff
@@ -332,3 +332,97 @@ using something similar to the following:
...
+   nn
+
+   # Step 1: Create an index with the "parent" columns to be included in the final Dataframe
```
Comment (jreback) [inline doc/source/gotchas.rst]: you can use sphinx to number these

File: doc/source/groupby.rst [hunk 6]
```diff
@@ -955,6 +959,42 @@ will be (silently) dropped. Thus, this does not pose any problems:
 
    df.groupby('A').std()
 
+.. note::
```
Comment (jreback) [inline doc/source/groupby.rst]: this is for another issue?

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/groupby.rst (hunk 6 above)
Comment (jreback) [reply doc/source/groupby.rst]: PR #18953 ?

Comment (pdpark) [reply doc/source/groupby.rst]: Yes, I should have made this a separate branch on my fork and separate pull request.

I will make the updates per your notes above.

**Comment by jreback** [comment]
superseded by #19175

File: doc/source/groupby.rst (hunk 6 above)
Comment (pdpark) [reply doc/source/groupby.rst]: I created a clean pull request for this fix: #19215

**Pull request closed without merging**